kernel/sched.c

```diff
@@ -767,14 +767,44 @@ bool z_thread_prio_set(struct k_thread *thread, int prio)
 		need_sched = z_is_thread_ready(thread);
 
 		if (need_sched) {
+#if defined(CONFIG_SMP)
```

Comment:
Style: "#ifdef CONFIG_..." is more idiomatic, "#if CONFIG_..." works too.  The use of defined() should be limited to expressions that need to compute boolean expressions, and even then is harder to read.

Reply:
Will update.